grammars/go.cson

```diff
@@ -162,19 +169,37 @@
         "match": "(\\+|&|\\+=|&=|&&|==|!=|\\-|\\||\\-=|\\|=|\\|\\||<|<=|\\*|\\^|\\*=|\\^=|<\\-|>|>=|/|<<|/=|<<=|\\+\\+|=|:=|;|%|>>|%=|>>=|\\-\\-|!|\\.\\.\\.|&\\^|&\\^=)"
       }
       {
-        "name": "keyword.operator.bracket.go",
-        "match": "(\\(|\\)|\\[|\\]|\\{|\\})"
+        'match': '\\{|\\}'
+        'name': 'meta.brace.curly.go'
       }
       {
-        "name": "keyword.operator.punctuation.go",
-        "match": "(\\.|,|:)"
+        'match': '\\(|\\)'
+        'name': 'meta.brace.round.go'
+      }
+      {
+        'match': '\\[|\\]'
+        'name': 'meta.brace.square.go'
       }
     ]
   'variables':
     'patterns': [
       {
-        'name': 'variable.go'
-        'match': '\\b[\\w_][\\w\\d_]*\\b',
+        'match': '\\b[\\w_][\\w\\d_]*\\b'
```

Comment:
Hmm... Yes, that is true. While I understand that it's a lot more work, it would be really neat to have a highlighter which is context aware.